text/353-cfn-registry-constructs.md

```diff
@@ -0,0 +1,467 @@
+all the resources and modules in the public CloudFormation registry.
+
+### Why should I use this feature?
+
+This library makes it easier to discover and use CloudFormation extensions in CDK apps.
+
+Previously, in order to use a resource from the CloudFormation registry, you would need to look up the resource documentation
+and use the low-level, weakly-typed `CfnResource` class to define it in your CDK app:
+
+```ts
+new CfnResource(this, 'MyEksCluster', {
+  type: 'AWSQS::EKS::Cluster',
+
+  // weakly-typed!
+  properties: {
+    name: 'my-new-cluster',
+    tags: [ { key: 'foo', value: 'bar' } ],
+  },
+});
+```
+
+With `@cdk-cloudformation-extensions` all public resources and modules will be listed in the **Construct Hub** like any
+other construct and by importing the relevant `cdk-cloudformation-extensions` module into their projects, they will be able to
+use them via strongly-typed classes. IDEs will show type information and inline help derived from the
+extension schema.
+
+```ts
+import { Cluster } from '@cdk-cloudformation-extensions/awsqs-eks';
+
+new Cluster(this, 'MyEksCluster', {
+  name: 'my-new-cluster',
+  tags: [ { key: 'foo', value: 'bar' } ],
+});
+```
+
+## Internal FAQ
+
+### Why are we doing this?
+
+We are doing this because CDK users would like to be able to use CloudFormation extensions 
+(modules, resources) as first-class citizens in CDK code. Extensions and native resource types
+should look and feel the same way when defined in CDK apps.
+
+Additionally, we would like to surface CloudFormation extensions in the upcoming Construct Hub
+and to that end, if we simply publish a construct library that includes constructs for all the public,
+extensions, this library will naturally be discoverable through the Hub.
+
+### Why should we _not_ do this?
+
+We have a longer term plan ([RFC#77](https://github.com/aws/aws-cdk-rfcs/issues/77)) to add 
+support for a CLI command called `cdk import` which will allow users to generate L1s from any
+registry type just-in-type and add them to their project (this is similar to CDK8s and CDKtf).
+
+### What is the technical solution (design) of this feature?
+
+The idea is to do something similar to what we do with the CloudFormation L1s 
+in the AWS CDK and simply publish JSII construct libraries which includes 
+statically-generated L1s for all the public registry extensions.
+
+#### Library per type
+
+CloudFormation extensions are semantically versioned. This means that new versions of an extension
+may include breaking changes. We also want to allow users to pick up a specific version of an extension.
+In light of these constraints, we determined that the best path forward is to publish each extension
+as a separate module, with a version that corresponds to the extension version in the CloudFormation
+registry. This way, we will have 1:1 alignment and users will have the most freedom.
```

Comment:
Does the version alignment include all levels (major, minor, and patch numbers)? What if we needed to apply some kind of security patch etc.?

Reply:
Good call. What do you think? I believe we can technically decide that the patch number is not coupled with the registry version to allow ourselves to deal with such cases.

Reply:
I think that would be acceptable.